components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java

```diff
@@ -379,10 +382,36 @@ public static void deleteTenant(String tenantDomain) throws Exception {
             tenantManager.deleteTenant(tenantId);
             log.info(String.format("Deleted tenant with domain: %s and tenant id: %d from the system.", tenantDomain,
                     tenantId));
-            triggerPostTenantDelete(tenantId, tenant.getTenantUniqueID(), tenant.getAdminUserId());
+            triggerPostTenantDelete(tenantId, tenantUuid, adminUserUuid);
         }
     }
 
+    private static String getTenantAdminUuid(Tenant tenant) throws UserStoreException {
+
+        String adminUserUuid = tenant.getAdminUserId();
+        if (StringUtils.isNotBlank(adminUserUuid)) {
+            return adminUserUuid;
+        }
+        // If the admin user uuid is not in the tenant object, we need to get it from user store.
+        String adminUsername = tenant.getAdminName();
+        try {
+            UserStoreManager userStoreManager = (UserStoreManager) TenantMgtServiceComponent.getRealmService().
+                    getTenantUserRealm(tenant.getId()).getUserStoreManager();
+            adminUserUuid = ((AbstractUserStoreManager) userStoreManager).getUserIDFromUserName(adminUsername);
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            log.error(
+                    String.format("Error occurred while getting user store manager for tenantId: %s", tenant.getId())
+            );
+        }
+        if (StringUtils.isBlank(adminUserUuid)) {
+            throw new UserStoreException(
+                    String.format("Error occurred while getting admin user unique identifier " +
```

Comment:
Shall we say `No UUID found for the admin user of tenant: %s`. Also lets throw an server exception here

Reply:
fixed